### PR TITLE
feat: add config for vim keybindings

### PR DESCRIPTION
**Pull request by mscolnick**

### CODE REVIEW

**Review by akshayka — COMMENTED** [review]
Nice! As a vim user I love this.

One comment so far.

- Delete cell shortcut (shift backspace) is not working.
- When cell is empty, perhaps `dd` should delete it. As a vim user that would feel natural to me.

**Review by akshayka — COMMENTED** [review]
I played around with this some more and noticed a bug (macOS, Chrome):

- keymap is set to vim
- in normal mode on a cell, vim works
- once I start editing (insert mode), I can never return to normal vim navigation mode. 

Perhaps this is because `escape` is required to exit insert mode, but instead `esc` just unfocuses the cell? Not sure ...

**Comment by mscolnick** [comment]
thanks for debugging the issue. i think it was from the more recent commit when i changed the ordering. 

with your suggestion and another change, this should be fixed. play around with it and looks pretty good. 
im am going to look if i can add a test for this (in another PR)

**Review by akshayka — COMMENTED** [review]
one final comment. a small bug related to `doubleCharacterListener`

File: frontend/src/core/codemirror/keymaps/keymaps.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          // Was handled
+          if (result) {
+            lastKey = "";
+            lastKeyTime = 0;
+            return true;
+          }
+        }
+
+        // Track keypress
+        lastKey = key;
```
[the remark this answers is not in the document]
Comment (mscolnick) [reply frontend/src/core/codemirror/keymaps/keymaps.ts:96]: fixed!

**Review by akshayka — APPROVED** [review]
Nice!!

**Pull request merged**